### PR TITLE
feat: Add Content-Type and Content-Length on request

### PR DESCRIPTION
**Pull request by victoraldir**
## Description

Slack's API documentation states that for JSON-encoded bodies client must explicitly set the **Content-type** HTTP header to `application/json`. This function does not explicitly set it. 

## Motivation and Context

We have internally created a wrapper of slack's api. Basically one single app is used by many internal purposes. We had followed strictly the API definition of Slack, including the enforcement of the **Content-type**. Turns out that this function didn't work seamlessly with this broker, and the reason was the  **Content-type**.

## Breaking Changes

This change does not affect neither the current nor latest versions of this module

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

I made use of the same **notify_slack_test.py** against a personal channel of Slack. All kind of message went through well

### CODE REVIEW

File: functions/notify_slack.py [hunk 1]
```diff
@@ -97,8 +97,10 @@ def notify_slack(subject, message, region):
     payload['text'] = "AWS notification"
     payload['attachments'].append(default_notification(subject, message))
 
-  data = urllib.parse.urlencode({"payload": json.dumps(payload)}).encode("utf-8")
+  data = json.dumps(payload).encode('utf-8')
```
Comment (antonbabenko) [inline functions/notify_slack.py:100]: `payload` key is missing, right?

**Review by antonbabenko — COMMENTED** [review]
I need someone else to take a look at this and confirm that the new code actually works.

File: functions/notify_slack.py (hunk 1 above)
Comment (AndrewFarley) [reply functions/notify_slack.py:100]: Also, the lack of urlencode suddenly, will this break something?  Was/is this not needed with the new headers set?

Comment (victoraldir) [reply functions/notify_slack.py:100]: @antonbabenko In spite of it works with `payload` as root key, this isn't clearly specified in the documentation. 
https://api.slack.com/messaging/webhooks

Comment (victoraldir) [reply functions/notify_slack.py:100]: Maybe `Content-Length` might not be really necessary, as urlib documentation states that is set when data is sent. 

On other hand, when `Content-Type` is not informed, it is added as `application/x-www-form-urlencoded` as default, which diverges of `application/json` required by Slack

https://docs.python.org/3/library/urllib.request.html#urllib.request.Request

**Pull request closed without merging**